scipy/signal/_spectral_py.py

```diff
@@ -1932,24 +1963,27 @@ def detrend_func(d):
             # Last point is unpaired Nyquist freq point, don't double
             result[..., 1:-1] *= 2
 
-    time = np.arange(nperseg/2, x.shape[-1] - nperseg/2 + 1,
+    time = xp.arange(nperseg/2, x.shape[-1] - nperseg/2 + 1,
                      nperseg - noverlap)/float(fs)
     if boundary is not None:
         time -= (nperseg/2) / fs
 
-    result = result.astype(outdtype)
+    result = xp.asarray(result, dtype=outdtype)
```

Comment:
`xp.astype(...)` is clearer

Reply:
This one was intentional because `astype` fails with `E   AttributeError: 'numpy.ndarray' object has no attribute '_array'`. I didn't look into it any deeper, but it is for the `numpy.array_api` backend.

Reply:
hmm maybe `result` is arriving as a regular `np` array? IIUC it should be a `np.array_api` array so `astype` should work.